File: specification/deploymentmanager/resource-manager/Microsoft.DeploymentManager/preview/2019-11-01-preview/examples-js/servicetopology_createorupdate.js
const { AzureDeploymentManager } = require("@azure/arm-deploymentmanager");
const { DefaultAzureCredential } = require("@azure/identity");

/**
 * This sample demonstrates how to Synchronously creates a new service topology or updates an existing service topology.
 *
 * @summary Synchronously creates a new service topology or updates an existing service topology.
 * x-ms-original-file: specification/deploymentmanager/resource-manager/Microsoft.DeploymentManager/preview/2019-11-01-preview/examples/servicetopology_createorupdate.json
 */
async function createATopologyWithArtifactSource() {
  const subscriptionId = "caac1590-e859-444f-a9e0-62091c0f5929";
  const resourceGroupName = "myResourceGroup";
  const serviceTopologyName = "myTopology";
  const serviceTopologyInfo = {
    artifactSourceId: "Microsoft.DeploymentManager/artifactSources/myArtifactSource",
    location: "centralus",
    tags: {},
  };
  const credential = new DefaultAzureCredential();
  const client = new AzureDeploymentManager(credential, subscriptionId);
  const result = await client.serviceTopologies.createOrUpdate(
    resourceGroupName,
    serviceTopologyName,
    serviceTopologyInfo
  );
  console.log(result);
}

createATopologyWithArtifactSource().catch(console.error);

File: specification/deploymentmanager/resource-manager/Microsoft.DeploymentManager/preview/2019-11-01-preview/examples-java/servicetopology_createorupdate.java
import java.util.HashMap;
import java.util.Map;

/** Samples for ServiceTopologies CreateOrUpdate. */
public final class Main {
    /*
     * x-ms-original-file: specification/deploymentmanager/resource-manager/Microsoft.DeploymentManager/preview/2019-11-01-preview/examples/servicetopology_createorupdate.json
     */
    /**
     * Sample code: Create a topology with Artifact Source.
     *
     * @param manager Entry point to DeploymentManager.
     */
    public static void createATopologyWithArtifactSource(
        com.azure.resourcemanager.deploymentmanager.DeploymentManager manager) {
        manager
            .serviceTopologies()
            .define("myTopology")
            .withRegion("centralus")
            .withExistingResourceGroup("myResourceGroup")
            .withTags(mapOf())
            .withArtifactSourceId("Microsoft.DeploymentManager/artifactSources/myArtifactSource")
            .create();
    }

    @SuppressWarnings("unchecked")
    private static <T> Map<String, T> mapOf(Object... inputs) {
        Map<String, T> map = new HashMap<>();
        for (int i = 0; i < inputs.length; i += 2) {
            String key = (String) inputs[i];
            T value = (T) inputs[i + 1];
            map.put(key, value);
        }
        return map;
    }
}
